README.md

```diff
@@ -0,0 +1,155 @@
+# [drf-paginator](https://github.com/yola/drf-paginator)
+
+JavaScript module for consuming paginated Django REST framework endpoints.
+
+[![npm Version](https://img.shields.io/npm/v/drf-paginator.svg?style=flat-square)](https://www.npmjs.com/package/drf-paginator)
+[![MIT License](https://img.shields.io/badge/license-MIT-blue.svg?style=flat-square)](https://github.com/yola/drf-paginator/master/LICENSE)
+[![Build Status](https://img.shields.io/travis/yola/drf-paginator.svg?style=flat-square)](https://travis-ci.org/yola/drf-paginator)
+[![Dependencies Status](https://img.shields.io/david/yola/drf-paginator.svg?style=flat-square)](https://www.npmjs.com/package/drf-paginator)
+[![devDependencies Status](https://img.shields.io/david/dev/yola/drf-paginator.svg?style=flat-square)](https://www.npmjs.com/package/drf-paginator)
+
+## Examples
+
+### Creating a paginator and fetching the first page:
+
+```javascript
+import drfp from 'drf-paginator';
+
+
+const paginator = drfp.paginate(request);
+
+paginator.first()
+  .then(function(response) {
+     // Do something with the response
+  });
+```
+
+### Creating a paginator using limit-offset pagination style:
+
+```javascript
+import {Paginator} from 'drf-paginator';
+
+
+const limit = 50;
+const style = Paginator.styles.limitOffset;
+const paginator = new Paginator(request, null, {limit}, {style});
+```
+
+### Fetching all of the results for a given request:
+
+```javascript
+import drfp from 'drf-paginator';
+
+
+drfp.all(request)
+  .then(function(response) {
+     // Do something with the response
+  });
+```
+
+### Using two paginators to iterate over ten pages at a time:
+
+```javascript
+import {PageMerger, Paginator} from 'drf-paginator';
+
+
+const paginator = new Paginator(request);
+const pageMerger = new PageMerger(paginator);
+
+const tenPageRequest = function(options, queryParams) {
+  const {page} = queryParams;
+  const startPage = (page * 10) - 9;
+  const endPage = ((page + 1) * 10) - 10;
+
+  return pageMerger.merge(startPage, endPage);
+};
+
+const tenPagePaginator = Paginator(tenPageRequest);
+```
+
+## Options
+
+* `limitQueryParam`
+  - Matches the endpoint's [`limit_query_param`](http://www.django-rest-framework.org/api-guide/pagination/#configuration_1) option
+  - Defaults to `'limit'`
+* `offsetQueryParam`
+  - Matches the endpoint's [`offset_query_param`](http://www.django-rest-framework.org/api-guide/pagination/#configuration_1) option
+  - Defaults to `'offset'`
+* `pageQueryParam`
+  - Matches the endpoint's [`page_query_param`](http://www.django-rest-framework.org/api-guide/pagination/#configuration) option
+  - Defaults to `'page'`
+* `style`
+  - Indicates which pagination style to use.
+  - Currently supports:
+    - [`PageNumberPagination`](http://www.django-rest-framework.org/api-guide/pagination/#pagenumberpagination)
+    - [`LimitOffsetPagination`](http://www.django-rest-framework.org/api-guide/pagination/#limitoffsetpagination)
```

Comment:
Testing our endpoints, do we support `LimitOffsetPagination` anywhere? Supporting alternate pagination styles seems like a feature that could come after the initial release.